core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,387 @@ export class ExplorationEditor extends BaseUser {
+      }
+    } else {
+      throw new Error(
+        `Expected next page button to be ${status}, but it was ${isDisabled ? 'disabled' : 'enabled'}`
+      );
+    }
+  }
+
+  /**
+   * Selects two revisions of an exploration for comparison.
+   * @param {number} revisionIndex1 - The index of the first revision to select.
+   * @param {number} revisionIndex2 - The index of the second revision to select.
+   */
+  async selectTwoRevisionsForComparison(
+    revisionIndex1: number,
+    revisionIndex2: number
+  ): Promise<void> {
+    if (revisionIndex1 === revisionIndex2) {
+      throw new Error('Both revision indexes cannot be the same.');
+    }
+
+    await this.page.waitForSelector(firstRevisionDropdown);
+    await this.clickOn(firstRevisionDropdown);
+    const panel1 = '#mat-select-0-panel';
+    await this.page.waitForSelector(
+      `${panel1} mat-option:nth-last-child(${revisionIndex1})`
+    );
+    await this.clickOn(
+      `${panel1} mat-option:nth-last-child(${revisionIndex1})`
+    );
+
+    await this.page.waitForSelector(secondRevisionDropdown);
+    await this.clickOn(secondRevisionDropdown);
+    const panel2 = '#mat-select-2-panel';
+    await this.page.waitForSelector(
+      `${panel2} mat-option:nth-last-child(${revisionIndex2})`
+    );
+    await this.clickOn(
+      `${panel2} mat-option:nth-last-child(${revisionIndex2})`
+    );
+  }
+
+  /**
+   * Verifies if changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectChangesInProperties(properties: string[]): Promise<void> {
+    const values: {[key: string]: string[]} = {};
+
+    for (let property of properties) {
+      await this.page.waitForSelector('.cm-atom', {visible: true});
+      const elements = await this.page.$$('.cm-atom');
+
+      const propertyElements = await Promise.all(
+        elements.map(async el => {
+          const textContent = await el.evaluate(node => node.textContent);
+          return textContent === property ? el : null;
+        })
+      );
+
+      const filteredElements = propertyElements.filter(el => el !== null);
+
+      if (filteredElements.length === 0) {
+        throw new Error(`Property ${property} not found`);
+      }
+
+      values[property] = (
+        await Promise.all(
+          filteredElements.map(async el => {
+            if (el) {
+              return el.evaluate(el => {
+                if (
+                  el.nextElementSibling &&
+                  el.nextElementSibling.nextElementSibling
+                ) {
+                  return el.nextElementSibling.nextElementSibling.textContent;
+                } else {
+                  throw new Error(
+                    `Unable to find third sibling for property ${property}`
+                  );
+                }
+              });
+            } else {
+              throw new Error(`Element is null for property ${property}`);
```

Comment:
1. What does "Element" refer to?
2. What does it mean for an element to be `null` for a given property?

Reply:
Updates, no longer needed.